### ISSUE TITLE
NNPACK condition should be changed (ARM processors)

### ISSUE THREAD

**Issue by snippler** [pytorch/pytorch#26396]
## 🚀 Feature
Remove the condition for usage of NNPACK for conv2d which requires the batchsize >= 16. Otherwise THNN is used on ARM processors

## Motivation

I tested NNPACK vs. THNN and I am think that NNPACK is mostly faster than THNN for batchsize = 1 and also > 1. NNPACK is still being excluded in special cases even when the condition batchsize >= 16 is removed, for example THNN is used when stride > 1.

The condition now makes little sense because it abolishes  NNPACK for inference (where batchsize = 1) where the speedup from NNPACK is most significant. In particular the speed-up is most significant because in case batchsize = 1 NNPACK uses a specific inference forward function which gives a an additional huge advantage in speed while for batchsize > 1 NNPACK uses a training forward function which is advantagious for backprop (I think at least).

To summarize: Best case scenario for inference  (usually batchsize = 1) - condition on NNPACK is removed and if other conditions allow NNPACK usage, NNNPACK is used and goes into sped-up inference function (only for batchsize = 1) and forward pass is super fast.

## Pitch

Remove condition in
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L173
input.size(0) >= 16

## Alternatives

Allow for use of Eigen Tensor Library and/or ARM Compute Library

**Comment by ezyang** [comment]
This seems reasonable. Feel free to mark me as reviewer for a PR.
